zulipterminal/model.py

```diff
@@ -824,13 +824,41 @@ def _get_stream_by_id(self, streams: List[StreamData], stream_id: int
                 return stream
         raise RuntimeError("Invalid stream id.")
 
+    def _unsubscribe_to_streams(self,
+                                stream_info_list: List[Subscription]) -> None:
+        is_pinned_stream_removed = False
+        is_unpinned_stream_removed = False
+        for stream_info in stream_info_list:
+            stream_id = stream_info['stream_id']
+            if self.is_stream_in_list(stream_id, self.pinned_streams):
+                stream = self._get_stream_by_id(self.pinned_streams,
+                                                stream_id)
+                self.pinned_streams.remove(stream)
+                is_pinned_stream_removed = True
+            else:
+                stream = self._get_stream_by_id(self.unpinned_streams,
+                                                stream_id)
+                self.unpinned_streams.remove(stream)
+                is_unpinned_stream_removed = True
+            if stream_id in self.muted_streams:
+                self.muted_streams.remove(stream_id)
+
+        if is_pinned_stream_removed:
+            sort_streams(self.pinned_streams)
+        if is_unpinned_stream_removed:
+            sort_streams(self.unpinned_streams)
```

Comment:
I know we improved this code in the past, and it does read well, but... if we just remove elements from a sorted container, isn't it still sorted?

Reply:
Ahh 😅 , I should have thought that. Thanks for pointing it out, would remove it shortly.